### DISCUSSION TITLE
can´t find the ko file after compiling

### DISCUSSION THREAD

**Discussion by danny1209** [OE4T/tegra-demo-distro#210] [Q&A] [answered]
Hi all, 
I created a meta-layer, which looks like as followed:
meta-test
----------recipes-test-example
---------------------------------test-example
-----------------------------------------------files
---------------------------------------------------aa.h, aa.c, bb.h, bb.c, Kbuild, Makefile
-----------------------------------------------test-example_0.1.bb

in the test-example_0.1.bb: RPROVIDES_${PN} += "kernel-module-test-example"
in the Kbuild: obj-m := test-example.o
test-example-y := aa.o bb.o
Makefile is standard: SRC := $(shell pwd)

                              all:
                              $(MAKE) -C $(KERNEL_SRC) M=$(SRC)
                              
                              modules_install:
                              $(MAKE) -C $(KERNEL_SRC) M=$(SRC) modules_install
                              
                              clean:
                              rm -f *.o *~ core .depend .*.cmd *.ko *.mod.c
                              rm -f Module.markers Module.symvers modules.order
                              rm -rf .tmp_versions Modules.symvers
in build/conf/bblayers.conf, this layer is also added.
in build/conf/local.conf, IMAGE_INSTALL_append += "test-example", also tried CORE_IMAGE_EXTRA_INSTALL += "test-example"

The result is: bitbake test-example, NO ERROR, everything is fine. bitbake demo-image-full also fine, NO ERROR.
but I can not find the test-example.ko fine anywhere.... did I do something wrong? thanks a lot for any help!

**Comment by ichergui** [comment, marked as answer]
Hey @danny1209 
Which branch you are using ?
Check your build directory if you can found the the kernel module *.ko
If exists you should add kernel-module-test-example in the IMAGE_INSTALL and not test-example only

**Reply by danny1209** [reply]
Hi @ichergui 
Thanks a lot. that´s one issue. another one is, I didn´t put the Kbuild file in bb file.

I´m using 32.6.1
